packages/twenty-front/src/modules/settings/security/hooks/useCreateSSOIdentityProvider.ts

```diff
@@ -0,0 +1,61 @@
+import { useSetRecoilState } from 'recoil';
+import {
+  CreateOidcIdentityProviderMutationVariables,
+  CreateSamlIdentityProviderMutationVariables,
+  useCreateOidcIdentityProviderMutation,
+  useCreateSamlIdentityProviderMutation,
+} from '~/generated/graphql';
+import { SSOIdentitiesProvidersState } from '@/settings/security/states/SSOIdentitiesProviders.state';
+
+export const useCreateSSOIdentityProvider = () => {
+  const [createOidcIdentityProviderMutation] =
+    useCreateOidcIdentityProviderMutation();
+  const [createSamlIdentityProviderMutation] =
+    useCreateSamlIdentityProviderMutation();
+
+  const setSSOIdentitiesProviders = useSetRecoilState(
+    SSOIdentitiesProvidersState,
+  );
+
+  const createSSOIdentityProvider = async (
+    input:
+      | ({
+          type: 'OIDC';
+        } & CreateOidcIdentityProviderMutationVariables['input'])
+      | ({
+          type: 'SAML';
+        } & CreateSamlIdentityProviderMutationVariables['input']),
+  ) => {
+    if (input.type === 'OIDC') {
+      // eslint-disable-next-line unused-imports/no-unused-vars
```

Comment:
can we avoid disabling the linter?

Reply:
I can do that, but I'll need to implement an omit function. It will be more complex than a simple restructuring.